apps/managedidentity/managedidentity_test.go

```diff
@@ -4,43 +4,120 @@ package managedidentity
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
 	"testing"
+	"time"
+
+	internalTime "github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/json/types/time"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/oauth/ops/accesstokens"
 )
 
-func fakeClient(mangedIdentityId ID, options ...Option) (Client, error) {
-	client, err := New(mangedIdentityId, options...)
+func fakeMIClient(mangedIdentityId ID, options ...ClientOption) (Client, error) {
+	fakeClient, err := New(mangedIdentityId, options...)
 
 	if err != nil {
 		return Client{}, err
 	}
 
-	return client, nil
+	return fakeClient, nil
+}
+
+type errorClient struct{}
+
+func (*errorClient) CloseIdleConnections() {}
+func (*errorClient) Do(req *http.Request) (*http.Response, error) {
+	return nil, fmt.Errorf("expected no requests but received one for %s", req.URL.String())
+}
+
+type FakeClient struct {
+	responseType int
+}
+
+func (c *FakeClient) CloseIdleConnections() {}
+func (c *FakeClient) Do(req *http.Request) (*http.Response, error) {
+	println(c.responseType)
+	w := makeResponse(c.responseType)
+	return &w, nil
+}
+
+func makeResponse(responseType int) http.Response {
+	if responseType == 1 {
+		return http.Response{
+			StatusCode: http.StatusOK,
+			Body: io.NopCloser(strings.NewReader(`{
+      "access_token": "fakeToken",
+      "refresh_token": "",
+      "expires_in": "3599",
+      "expires_on": "1506484173",
+      "not_before": "1506480273",
+      "token_type": "Bearer"
+    }`)),
+			Header: make(http.Header),
+		}
+	} else {
+		return http.Response{
+			StatusCode: http.StatusBadRequest,
+			Body:       io.NopCloser(strings.NewReader(`{}`)),
+			Header:     make(http.Header),
+		}
+	}
 }
+func TestManagedIdentityIMDS_SAMISuccess(t *testing.T) {
+	fakeHTTPClient := FakeClient{responseType: 1}
 
-func TestManagedIdentity(t *testing.T) {
-	client, err := fakeClient(SystemAssigned())
+	client, err := fakeMIClient(SystemAssigned(), WithHTTPClient(&fakeHTTPClient))
```

Comment:
Consider ways to make these tests more readable. For example this private function doesn't really seem to be needed. A bit of dup code to improve readability is preferred.